### PR TITLE
[DC-1543] convert storage script to cleaning rule

### PR DESCRIPTION
**Pull request by lrwb-aou**
* converts the previous storage script to a multi-stage cleaning rule
* stores the pids and generated shifts in a sandbox before appending to primary pid/rid table.  this will allow more identification if something unexpected happens.

### CODE REVIEW

File: data_steward/cdr_cleaner/cleaning_rules/store_pid_rid_mappings.py [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+  , CAST(FLOOR({{max_shift}} * RAND() + 1) AS INT64) as shift
+FROM `{{rdr_table.project}}.{{rdr_table.dataset_id}}.{{rdr_table.table_id}}`
+WHERE person_id not in (
+  SELECT person_id
+  FROM `{{primary.project}}.{{primary.dataset_id}}.{{primary.table_id}}`)
+-- This is just to make sure we don't duplicate either person_id OR research_id --
+AND research_id not in (
+  SELECT research_id
+  FROM `{{primary.project}}.{{primary.dataset_id}}.{{primary.table_id}}`)
+""")
```
Comment (dcampbell-vumc) [inline data_steward/cdr_cleaner/cleaning_rules/store_pid_rid_mappings.py:53]: Thanks for these comments!

**Review by dcampbell-vumc — CHANGES_REQUESTED** [review]
This all looks good, just a couple of nit picks. 

There isn't an integration test created for this yet so for me to test this, I printed out all the queries, popped them in BQ to at least see what data ended up populated in the created `stage_less_dc1543_pid_rid_mapping` sandbox table. It all looked right to me in the end!

**Comment by lrwb-aou** [comment]
> This all looks good, just a couple of nit picks.
> 
> There isn't an integration test created for this yet so for me to test this, I printed out all the queries, popped them in BQ to at least see what data ended up populated in the created `stage_less_dc1543_pid_rid_mapping` sandbox table. It all looked right to me in the end!

You're right.  The integration test will be a separate issue.  I've already started working on it.  👍

File: data_steward/cdr_cleaner/clean_cdr.py [hunk 2]
```diff
@@ -124,6 +125,7 @@
 ]
 
 RDR_CLEANING_CLASSES = [
+    (StoreNewPidRidMappings,),
```
[the remark this answers is not in the document]
Comment (lrwb-aou) [reply data_steward/cdr_cleaner/clean_cdr.py:128]: The order is not important for this rule.  It is reading a standalone table, writing to a sandbox table for good measure, then updating a primary table.  Do you have a suggestion for a better location?

File: data_steward/cdr_cleaner/cleaning_rules/store_pid_rid_mappings.py [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+priamry pid_rid_mapping table.
+
+These records will be appended to the pipeline_tables.pid_rid_mapping table in BigQuery.
+Duplicate mappings are not allowed.
+"""
+# Python imports
+import argparse
+import inspect
+import logging
+import time
```
[the remark this answers is not in the document]
Comment (lrwb-aou) [reply data_steward/cdr_cleaner/cleaning_rules/store_pid_rid_mappings.py:15]: Nice catch.  They should be removed.

File: data_steward/cdr_cleaner/clean_cdr.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (dcampbell-vumc) [reply data_steward/cdr_cleaner/clean_cdr.py:128]: Nope, this is fine. I remember for the CT cleaning rules, we had specific location comments for some cleaning rules.

File: data_steward/cdr_cleaner/cleaning_rules/store_pid_rid_mappings.py [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+primary pid_rid_mapping table.
+
+These records will be appended to the pipeline_tables.pid_rid_mapping table in BigQuery.
+Duplicate mappings are not allowed.
+"""
+# Python imports
+import argparse
+import inspect
+import logging
+import time
```
Comment (lrwb-aou) [inline data_steward/cdr_cleaner/cleaning_rules/store_pid_rid_mappings.py:15]:
```suggestion
import logging
```

**Comment by lrwb-aou** [comment]
@dcampbell-vumc I made some changes.  Let me know how they look.

**Review by dcampbell-vumc — APPROVED** [review]
This is good to go!

**Pull request merged**